examples/jetstream/api-migration/go/main.go

```diff
@@ -0,0 +1,255 @@
+	consumerName := "dur-1"
+	oldJS.AddConsumer(streamName, &nats.ConsumerConfig{
+		Name:              consumerName,
+		DeliverSubject:    "handler-1",
+		AckPolicy:         nats.AckExplicitPolicy,
+		InactiveThreshold: 10 * time.Minute,
+	})
+
+	// ### Retrieving messages synchronously with `SubscribeSync()`
+	// In order to retrieve messages synchronously, the `SubscribeSync()`
+	// method can be used. This method will create a push consumer
+	// and a subscription to receive messages.
+	// `SubscribeSync` can also be used to bind to existing streams, consumers
+	// etc. (similarly to `Subscribe`).
+	fmt.Println("# SubscribeSync")
+	sub, _ = oldJS.SubscribeSync("events.>", nats.AckExplicit())
+
+	// To *receive* a message, call `NextMsg` with a timeout. The timeout
+	// applies when pending count is zero and the consumer has fully caught
+	// up to the available messages in the stream. If no messages become
+	// available, this call will only block until the timeout.
+	msg, _ := sub.NextMsg(time.Second)
+	fmt.Printf("received %q\n", msg.Subject)
+	msg.Ack()
+
+	// ### Pull consumers
+	// The legacy API also supports pull consumers. However, these are
+	// greatly limited in functionality since it is only possible to pull
+	// a specific number of messages, without any optimization or prefetching.
+	// That makes using pull consumers in the legacy API inefficient in contrast
+	// to push consumers.
+	// In order to create a pull consumer and consume messages, the
+	// `PullSubscribe` method can be used.
+	fmt.Println("# Subscribe with pull consumer")
+	sub, _ = oldJS.PullSubscribe("events.>", "pull-cons", nats.AckExplicit())
+
+	// Messages can be retrieved using the `Fetch` or `FetchBatch` methods.
+	// `Fetch` will retrieve up to the provided number of messages and block
+	// until the either all messages are available or the timeout is reached.
+	fmt.Println("# Fetch")
+	msgs, _ := sub.Fetch(2, nats.MaxWait(100*time.Millisecond))
+	for _, msg := range msgs {
+		fmt.Printf("received %q\n", msg.Subject)
+		msg.Ack()
+	}
+
+	// `FetchBatch` is similar to `Fetch` except it will return a channel
+	// on which the messages will be delivered. This allows for processing
+	// messages as they become available.
+	fmt.Println("# FetchBatch")
+	msgs2, _ := sub.FetchBatch(2, nats.MaxWait(100*time.Millisecond))
+	for msg := range msgs2.Messages() {
+		fmt.Printf("received %q\n", msg.Subject)
+		msg.Ack()
+	}
+
+	// ## New JetStream API
+	// Now let's look at the new JetStream API for creating and managing
+	// streams and consumers.
+	fmt.Println("\n# New API")
+
+	// The new JetStream API is located in the `jetstream` package.
+	// In order to create an entry point to the JetStream API, use the
+	// `New` function.
+	newJS, _ := jetstream.New(nc)
+
+	// The new API uses `context.Context` for cancellation and timeouts when
+	// managing streams and consumers.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Having a single context for this example works fine since the runtime is less than the timeout, but in practice people should use individually defined contexts for a long running process. Maybe leave off the `WithTimeout` wrapper to avoid beginners from using this without realizing the side effect?